adagrams/game.py

```diff
@@ -1,11 +1,120 @@
+import random 
+
+LETTER_POOL = {
+    'A': 9, 
+    'B': 2, 
+    'C': 2, 
+    'D': 4, 
+    'E': 12, 
+    'F': 2, 
+    'G': 3, 
+    'H': 2, 
+    'I': 9, 
+    'J': 1, 
+    'K': 1, 
+    'L': 4, 
+    'M': 2, 
+    'N': 6, 
+    'O': 8, 
+    'P': 2, 
+    'Q': 1, 
+    'R': 6, 
+    'S': 4, 
+    'T': 6, 
+    'U': 4, 
+    'V': 2, 
+    'W': 2, 
+    'X': 1, 
+    'Y': 2, 
+    'Z': 1
+}
+
 def draw_letters():
-    pass
+    available_letters = []
+    ten_random_letters = []
+
+    # I am iterating through each letter and number in the dictionary so that I can
+    # get a list (available_letters) showing all available letters. 
+    for letter, number in LETTER_POOL.items():
+        available_letters.append(letter)
+    
+
+    while len(ten_random_letters) < 10:
+        one_random_letter = random.choice(available_letters)
+        ten_random_letters.append(one_random_letter)
+        available_letters.remove(one_random_letter)
```

Comment:
**Selecting multiples**: This isn't *quite* right. When you remove a randomly-selected letter in this way, it makes it so that you never draw a letter more than once! To see this in action, you can add this test to the end of your wave 1 tests:

```python
def test_ever_draws_duplicate_letters():
    def has_dupes(letter_bank):
        unique_letters = set(letter_bank)
        return len(unique_letters) != len(letter_bank)

    saw_a_duplicate = False
    for i in range(1000):
        letters = draw_letters()
        saw_a_duplicate = saw_a_duplicate or has_dupes(letters)

    assert saw_a_duplicate
```

You want the likelihood of drawing any given letter to be based on the number of letters in the pool. Think of it like drawing ten letters out of a bag (the way you do with games like "bananagrams" or "scrabble") into your hand. When you sample from the list of `LETTER_POOL` keys, it's as though you only ever put each letter into the bag once! To get behavior like this, your loop that creates `available_letters` should have added the appropriate number of letters to it–e.g. it should have added 1 `"Z"`, but 12 `"E"`s!

---

**Regarding built-in functions** like `random.choice`: when bringing in functions from imported modules (that we haven’t covered in the curriculum), please consider including a comment about how you understand the function to work, and how you researched it. Now that we’ve discussed Big O, it’s also useful to think about what the time/space complexity of the function might be. Thinking about how we can implement this functionality ourselves can help us gain insight into what that might be.

Remember that not every language has the same built-in functions, so we should be able to implement this same functionality ourselves. In the projects, we don’t ask for any features that we don’t think you should be able to write yourself. For drawing a random hand, the only external function that would be “required” is a way to pick a random number (such as randint). At this stage in our coding journeys, it’s often more valuable to re-implement things by hand to practice thinking about how to break down and approach the problem than to use a single library function to solve the problem for us.